src/main/java/org/fcrepo/migration/validator/impl/ValidatingObjectHandler.java

```diff
@@ -0,0 +1,233 @@
+        for (final var dsVersion : dsVersions) {
+            if (dsVersion.isLastVersionIn(objectReference)) {
+                final var dsInfo = dsVersion.getDatastreamInfo();
+                if (!dsInfo.getState().equals("D")) {
+                    headDatastreamIds.add(dsId);
+                }
+
+                try {
+                    final var headers = this.ocflSession.readHeaders(targetResource);
+                    final String details;
+                    final var sourceValue = dsVersion.getCreated();
+                    final var targetValue = headers.getCreatedDate().toString();
+                    final var result = sourceValue.equals(targetValue) ? OK : FAIL;
+                    if (result.equals(OK)) {
+                        details = format("HEAD binary creation dates match: %s", sourceValue);
+                    } else {
+                        details = format("HEAD binary creation dates do no match: sourceValue=%s, target value=%s",
+                                sourceValue, targetValue);
+                    }
+                    validationResults.add(new ValidationResult(indexCounter++, result, OBJECT_RESOURCE,
+                            BINARY_METADATA, sourceOjbectId, targetObjectId, sourceResource, targetResource, details));
+                    validationResults.add(new ValidationResult(indexCounter++, OK, OBJECT_RESOURCE,
+                            SOURCE_OBJECT_RESOURCE_EXISTS_IN_TARGET, sourceOjbectId, targetObjectId, sourceResource,
+                            targetResource, "Source object resource exists in target."));
+                } catch (NotFoundException ex) {
+                    validationResults.add(new ValidationResult(indexCounter++, FAIL, OBJECT_RESOURCE,
+                            SOURCE_OBJECT_RESOURCE_EXISTS_IN_TARGET, sourceOjbectId, targetObjectId, sourceResource,
+                            targetResource, "Source object resource does not exist in target."));
+
+                }
+            }
+
+            sourceVersionCount++;
+        }
+
+        try {
+            final var targetVersionCount = this.ocflSession.listVersions(targetResource).size();
+            final var ok = sourceVersionCount == targetVersionCount;
+            var details = format("binary version counts match for resource: %s", sourceVersionCount);
+            if (!ok) {
+                details = format("binary version counts do not match: source=%d, target=%d", sourceVersionCount,
+                        targetVersionCount);
+            }
+            validationResults.add(new ValidationResult(indexCounter++, ok ? OK : FAIL, OBJECT_RESOURCE,
+                    BINARY_VERSION_COUNT, sourceOjbectId, targetObjectId, sourceResource, targetResource, details));
+        } catch (NotFoundException ex) {
+            // intentionally left blank: we check for existence above
+        }
+
+    }
+
+
+    private void completeObjectValidation(final ObjectInfo objectInfo) {
```

Comment:
The `objectInfo` argument is never used. Remove it?

Reply:
+1